nemo_aligner/utils/distributed.py

```diff
@@ -17,29 +17,46 @@
 import time
 import warnings
 from collections import defaultdict
+from contextlib import contextmanager
 from dataclasses import dataclass
 from datetime import timedelta
 from typing import Dict, Optional, Union
 
 import torch
-from megatron.core import parallel_state, tensor_parallel
+from megatron.core import tensor_parallel
 
-from nemo.collections.nlp.modules.common.text_generation_utils import get_model_parallel_src_rank
 from nemo.utils.timers import NamedTimer
+from nemo_aligner.utils import parallel_state
+from nemo_aligner.utils.parallel_state import get_model_parallel_group, get_model_parallel_src_rank
 from nemo_aligner.utils.ppo_utils import calculate_entropy
 
 
-def gather_tensor(tensor, dst, group, dtype=torch.float32):
-    """Gather any 2d tensor to the dst rank from every other rank in the given group.
-    All the ranks that send or receive data must call this function."""
-    tensor = tensor.to(device=torch.cuda.current_device(), dtype=dtype)
-    if torch.distributed.get_rank() == dst:
-        gather_list = [torch.empty_like(tensor) for _ in range(torch.distributed.get_world_size(group))]
-    else:
-        gather_list = None
+def rebalance_nd_tensor(tensor, group):
+    """takes tensors with variable leading sizes(at dim=0) and then stack them into a single tensor. NOTE: assumes dim != 0 to have all the same shape
```

Comment:
I believe this is what you mean?
```suggestion
    """
    Takes tensors with variable leading sizes (at dim=0) and then stack them into a single tensor.
    
    NOTE: assumes all other (i.e., non-zero) dimensions are equal.
```

Reply:
yup! thanks for making it clearer